keel-sql/src/main/resources/db/changelog/20200611-create-resource-metadata-view.yml

```diff
@@ -0,0 +1,30 @@
+databaseChangeLog:
+- changeSet:
+    id: create-resource-metadata-view
+    author: fletch
+    changes:
+    - createView:
+        viewName: resource_with_metadata
+        selectQuery: |
+          select
+            resource.uid,
+            resource.id,
+            resource.application,
+            resource.kind,
+            json_object(
+              'uid', resource.uid,
+              'id', resource.id,
+              'application', resource.application,
+              'environment', environment.uid,
+              'environmentName', environment.name,
+              'deliveryConfig', delivery_config.uid,
+              'serviceAccount', delivery_config.service_account
+            ) as metadata,
+            resource.spec
+          from resource
+          left outer join environment_resource on resource.uid = environment_resource.resource_uid
+          left outer join environment on environment.uid = environment_resource.environment_uid
+          left outer join delivery_config on delivery_config.uid = environment.delivery_config_uid
+    - dropColumn:
+        tableName: resource
+        columnName: metadata
```

Comment:
Instead of dropping it altogether, I think it might be prudent to allow resource-specific metadata to coexist with the metadata inherited from the delivery config and environment. I'd propose we merge the two together when reading. That would also avoid yet another discrepancy from the k8s format, which I get the feeling we're kinda regretting having diverged from.

Reply:
The reason I went for dropping it altogether is that we currently don't use it, and I thought it might be confusing for us if we don't realize that not all the metadata is in the table. This doesn't have any impact on the in/out format of resources.

Reply:
Ah, right. Makes sense!